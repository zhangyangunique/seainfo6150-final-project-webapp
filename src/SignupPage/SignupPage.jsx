import React, { useState } from "react";
import styles from "./SignupPage.module.css";

const SignupPage = () => {
	const [submittedForm, setSubmittedForm] = useState();
	function onSubmit(e) {
		e.preventDefault();
		const data = new FormData(e.target);
		setSubmittedForm(data);
	};

	return (
		<div>
			<h2>Sign Up Now</h2>
			{
				submittedForm ? (
					<div className={styles.registeredText}>Thank you {submittedForm.get("firstName")}! You have successfully registered with email address {submittedForm.get("email")}</div>
				) : (
						<form className={styles.signupForm} onSubmit={onSubmit}>
							<div className={styles.formGroup}>
								<label htmlFor="firstName">First Name</label>
								<input
									className={styles.formControl}
									type="text" name="firstName"
									required="required"
									pattern="[a-zA-Z]{1,20}"
									title="only letters, maximum is 20 characters"
									id="firstName"
								/>
							</div>
							<div className={styles.formGroup}>
								<label htmlFor="lastName">Last Name</label>
								<input
									className={styles.formControl}
									type="text"
									name="lastName"
									required="required"
									pattern="[a-zA-Z]{1,20}"
									title="only letters, maximum is 20 characters"
									id="lastName"
								/>
							</div>
							<div className={styles.formGroup}>
								<label htmlFor="userName">User Name</label>
								<input
									className={styles.formControl}
									type="text"
									name="userName"
									required="required"
									id="userName"
									pattern="[a-zA-Z0-9]{1,20}"
									title="only letters, numbers, maximum is 20 characters"
								/>
							</div>
							<div className={styles.formGroup}>
								<label htmlFor="email">Email Address</label>
								<input className={styles.formControl} type="email" name="email" required="required" id="email" />
							</div>
							<div className={styles.formGroup}>
								<label htmlFor="password">Password</label>
								<input
									className={styles.formControl}
									type="password" name="password"
									required="required"
									pattern="[\s\S]{6,30}"
									title="password length should between 6 and 30 characters"
									id="password" />
							</div>
							<div className={styles.formGroup}>
								<input className={`${styles.formControl} ${styles.submitButton}`} type="submit" value="Submit" />
							</div>
						</form>
					)
			}
		</div>
	)
}

export default SignupPage