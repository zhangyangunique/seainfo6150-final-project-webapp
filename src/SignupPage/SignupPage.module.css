.signupForm {
	display: block;
	width: 50%;
	margin: 0 auto;
	font-family: -apple-system, BlinkMacSystemFont, "San Francisco", "Helvetica Neue", Helvetica, Ubuntu, Roboto, Noto, "Segoe UI", Arial, sans-serif;
	font-size: 18px;
	font-weight: 400;
	line-height: 1.6;
	color: #111111;
}

.formGroup {
	display: block;
	margin-top: 1rem;
	width: 100%;
	float: left;
	position: relative;
	padding-left: 12px;
	padding-right: 12px;
	box-sizing: border-box;
}

div {
	display: block;
}

.formControl {
	display: block;
	width: 100%;
	height: 30px;
	padding: 6px 12px;
	font-size: 15px;
	line-height: 1.6;
	color: #111111;
	background-color: transparent;
	background-image: none;
	border: 1px solid #767676;
	border-radius: 3px;
}

label {
	display: inline-flex;
	-webkit-box-align: baseline;
	align-items: baseline;
	max-width: 100%;
	margin-bottom: 6px;
	flex-wrap: wrap;
}

.submitButton {
	font-size: 18px;
	width: 80%;
	height: 40px;
	margin: 0 auto;
	text-align: center;
	color: white;
	background-color: #111111;
	margin-bottom: 50px;
}

.registeredText {
	margin-top: 200px;
	text-align: center;
	font-size: 30px;
}

h2 {
	text-align: center;
	font-size: 30px;
	font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.gender label{
    display: inline-block;
    margin-right: 40px;
}

.country label {
    margin-right: 32px;
}

.profession label {
    margin-right: 10px;
}

.dropdownForm {
    width: 80%;
    box-sizing: border-box;
    border: 1px solid grey;
    border-radius: 4px;
    padding: 10px;
    margin-bottom: 10px;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}