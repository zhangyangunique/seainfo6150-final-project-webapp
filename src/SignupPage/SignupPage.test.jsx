import React from "react";
import { render } from "@testing-library/react";
import SignupPage from "./SignupPage";

describe("SignupPage tests", () => {
    it("renders correctly", () => {
        const { container } = render(<SignupPage />);
        expect(container).toMatchSnapshot();
    });
});