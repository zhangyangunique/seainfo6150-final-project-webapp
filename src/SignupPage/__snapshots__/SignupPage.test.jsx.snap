// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage tests renders correctly 1`] = `
<div>
  <div>
    <h2>
      Sign Up Now
    </h2>
    <form
      class="signupForm"
    >
      <div
        class="formGroup"
      >
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          class="formControl"
          id="firstName"
          name="firstName"
          pattern="[a-zA-Z]{1,20}"
          required=""
          title="only letters, maximum is 20 characters"
          type="text"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          class="formControl"
          id="lastName"
          name="lastName"
          pattern="[a-zA-Z]{1,20}"
          required=""
          title="only letters, maximum is 20 characters"
          type="text"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="userName"
        >
          User Name
        </label>
        <input
          class="formControl"
          id="userName"
          name="userName"
          pattern="[a-zA-Z0-9]{1,20}"
          required=""
          title="only letters, numbers, maximum is 20 characters"
          type="text"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="email"
        >
          Email Address
        </label>
        <input
          class="formControl"
          id="email"
          name="email"
          required=""
          type="email"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          class="formControl"
          id="password"
          name="password"
          pattern="[\\\\s\\\\S]{6,30}"
          required=""
          title="password length should between 6 and 30 characters"
          type="password"
        />
      </div>
      <div
        class="formGroup"
      >
        <div
          class="gender"
        >
          <label>
            Gender:
          </label>
          <label>
            Male
            <input
              checked=""
              id="male"
              name="gender"
              type="radio"
              value="male"
            />
          </label>
          <label>
            Female
            <input
              checked=""
              id="female"
              name="gender"
              type="radio"
              value="female"
            />
          </label>
          <label>
            Other
            <input
              checked=""
              id="other"
              name="gender"
              type="radio"
              value="other"
            />
          </label>
        </div>
      </div>
      <div
        class="formGroup"
      >
        <div
          class="country"
        >
          <label
            for="country"
          >
            Country:
          </label>
          <select
            class="dropdownForm"
            id="country"
            name="country"
          >
            <option
              value="usa"
            >
              United States
            </option>
            <option
              value="canada"
            >
              Canada
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
      </div>
      <div
        class="formGroup"
      >
        <div
          class="profession"
        >
          <label
            for="profession"
          >
            Profession:
          </label>
          <select
            class="dropdownForm"
            id="profession"
            name="profession"
          >
            <option
              value="student"
            >
              Student
            </option>
            <option
              value="teacher"
            >
              Teacher
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
      </div>
      <div
        class="formGroup"
      >
        <input
          class="formControl submitButton"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;
